### PR TITLE
No cluster

### LINKED ISSUE #57: Running without Cluster Manager
### Description

The bot currently doesn't run properly without using the cluster manager.

### Usefulness

This would allow contributors and self-hosting users to run the bot without the cluster manager running.

### Have You...?
- [x] I have checked previous issues to see if my enhancement or feature has already been suggested.
- [x] I have checked previous issues and pull requests to see if my enhancement or feature has already been completed for an upcoming release.

### LINKED ISSUE #58: Refactor Configuration
### Description

Changing the configuration file from JSON to env.

### Usefulness

It provides an easier way to retrieve configuration.

### Have You...?
- [x] I have checked previous issues to see if my enhancement or feature has already been suggested.
- [x] I have checked previous issues and pull requests to see if my enhancement or feature has already been completed for an upcoming release.

### PR DESCRIPTION
**Pull request by bwpikaard**
### Items Changed

- [x] Commands
- [x] Events
- [x] Handlers
- [ ] Structures
- [ ] Tasks
- [ ] Documentation
- [x] Other: package.json, /index.js, scripts/*, package-example.json

### Description

Adding a ClusterUtil to allow running without the cluster manager. Update from config.json to .env file for configuration.

### Issues

Closes #57.
Closes #58.

### Have You...?

- [x] I have read the [Contributing Guidelines](https://github.com/typicalbot/typicalbot/blob/master/.github/CONTRIBUTING.md).
- [x] I have completed the [pull request template](https://github.com/typicalbot/typicalbot/blob/master/.github/PULL_REQUEST_TEMPLATE.md).
- [x] I have checked open [pull requests](https://github.com/typicalbot/typicalbot/pulls) for upcoming features and fixes.

### CODE REVIEW

**Comment by tjrgg** [comment]
There's a couple of issues with this PR.

First, there's a couple of conflicts that need to be resolved. The best way to do this would be to rebase your branch. Second, removing a joke is outside the scope of your PR. Since the jokes are a separate issue, that should be addressed in a separate PR that's focused on the clustering and configuration.

**Comment by tjrgg** [comment]
I've went ahead and rebased the branch for you and reverted the commit removing the joke.

File: src/permissions/TypicalBotMaintainer.js [hunk 1]
```diff
@@ -12,7 +12,7 @@ class TypicalBotCreator extends PermissionLevelStr {
     }
 
     check(guild, member) {
-        return guild.client.config.maintainers.includes(member.id);
+        return process.env.MAINTAINERS.includes(member.id);
```
Comment (nsylke) [inline src/permissions/TypicalBotMaintainer.js:15]: Does `"123","456"` even work in the .env file and return an array?

File: index.js [hunk 2]
```diff
@@ -1,45 +1,48 @@
...
-const node = new Node(process.env.CLUSTER)
-    .on('error', (error, client) => console.error(`[IPC] Error from ${client.name}:`, error))
-    .on('socket.disconnect', (client) => console.error(`[IPC] Disconnected from ${client.name}`))
-    .on('socket.destroy', (client) => console.error(`[IPC] Client Destroyed: ${client.name}`))
-    .on('socket.ready', async (client) => {
-        console.log(`[IPC] Connected to: ${client.name}`);
-    });
 
-node.connectTo(config.nodePort).catch((error) => console.error('[IPC] Disconnected!', error));
+if (/^\s*(true|1|on|yes)\s*$/.test(process.env.USE_CLUSTER)) {
```
Comment (nsylke) [inline index.js:9]: What's the point of this if we are setting the variable to a boolean on L3-L4?

File: scripts/pm2.js [hunk 3]
```diff
@@ -1,10 +1,10 @@
-/* eslint-disable no-console */
 import { connect, start, disconnect } from 'pm2';
 import { Util } from 'discord.js';
-import { token, clusterServer, clusterBuild } from '../config';
+
+require('dotenv').config();
```
Comment (nsylke) [inline scripts/pm2.js:4]: Inconsistent with the above modules.

File: scripts/setup.js [hunk 4]
```diff
@@ -1,8 +1,12 @@
-/* eslint-disable no-console */
-import redb from 'rethinkdbdash';
-import { database } from '../config';
+require('dotenv').config();
+const redb = require('rethinkdbdash');
+
+const credentials = {
```
Comment (nsylke) [inline scripts/setup.js:4]: can we just remove this variable and put credentials right into the constructor on L10?

File: src/permissions/TypicalBotMaintainer.js (hunk 1 above)
Comment (tjrgg) [reply src/permissions/TypicalBotMaintainer.js:15]: No. That's a string method.

Comment (nsylke) [reply src/permissions/TypicalBotMaintainer.js:15]: Oh right. I'm not sure why I was thinking it was an array method.

File: index.js [hunk 5]
```diff
@@ -1,45 +1,48 @@
...
+    const node = new Node(process.env.CLUSTER)
+        .on('error', (error, client) => console.error(`[IPC] Error from ${client.name}:`, error))
+        .on('socket.disconnect', (client) => console.error(`[IPC] Disconnected from ${client.name}`))
+        .on('socket.destroy', (client) => console.error(`[IPC] Client Destroyed: ${client.name}`))
+        .on('socket.ready', async (client) => {
+            console.log(`[IPC] Connected to: ${client.name}`);
```
Comment (nsylke) [inline index.js:15]: Inconsistent with the other emitted events.

Comment (tjrgg) [reply index.js:15]: This PR doesn't change any of that. It only moves it.

File: index.js (hunk 2 above)
Comment (bwpikaard) [reply index.js:9]: env doesn't support booleans, so setting it doesn't do anything. I'll remove at some point.

**Comment by tjrgg** [comment]
#94 closes this PR.

**Pull request closed without merging**